RUFAS/biophysical/animal/nutrients/beef_cow_calf_requirements_calculator.py

```
@@ -0,0 +1,482 @@
+            raise ValueError("BEEF_BULL cannot have days_pregnant set; bulls do not gestate.")
+        if inputs.days_pregnant is not None and not (
+            1 <= inputs.days_pregnant <= AnimalModuleConstants.BEEF_GESTATION_LENGTH_DAYS
+        ):
+            raise ValueError(
+                f"days_pregnant must be 1–{AnimalModuleConstants.BEEF_GESTATION_LENGTH_DAYS}; "
+                f"got {inputs.days_pregnant}."
+            )
+        if inputs.days_in_milk is not None and inputs.days_in_milk < 0:
+            raise ValueError(f"days_in_milk must be non-negative; got {inputs.days_in_milk}.")
+        if inputs.body_weight <= 0.0:
+            raise ValueError(f"body_weight must be positive, got {inputs.body_weight}")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)